snc-library.sh

```diff
@@ -53,12 +54,17 @@ function run_preflight_checks() {
 
         echo "Checking libvirt and DNS configuration"
 
-        LIBVIRT_URI=qemu+tcp://localhost/system
+	LIBVIRT_URI=qemu:///system
+	if [ ${preset} == "snc" ]; then
+           LIBVIRT_URI=qemu+tcp://localhost/system
+	fi
 
         # check if libvirtd is listening on a TCP socket
-        if ! virsh -c ${LIBVIRT_URI} uri >/dev/null; then
-                preflight_failure  "libvirtd is not listening for plain-text TCP connections, see https://github.com/openshift/installer/tree/master/docs/dev/libvirt#configure-libvirt-to-accept-tcp-connections"
-        fi
+	if [ ${preset} == "snc" ]; then
+            if ! virsh -c ${LIBVIRT_URI} uri >/dev/null; then
+                    preflight_failure  "libvirtd is not listening for plain-text TCP connections, see https://github.com/openshift/installer/tree/master/docs/dev/libvirt#configure-libvirt-to-accept-tcp-connections"
+            fi
+	fi
```

Comment:
maybe better as `run_generic_preflight_checks()` and `run_snc_preflight_checks()`? Just suggestions, not sure which is better. Need to start moving snc to some higher level language, go or python ;)